cpp/src/arrow/ipc/message.h

```diff
@@ -24,6 +24,7 @@
 #include <string>
 #include <utility>
 
+#include "arrow/io/interfaces.h"
```

Comment:
We can forward-declare io::RandomAccessFile if we want to avoid including the entire header.

Reply:
I remember it is `io:ReadRange` that makes me add this header. I've forward-declare it in the newly added `reader_internal.h` now.